plugins/discovery-file/src/test/java/org/elasticsearch/discovery/file/FileBasedUnicastHostsProviderTests.java

```diff
@@ -152,13 +153,20 @@ public void testSomeInvalidHostEntries() throws Exception {
         final Settings settings = Settings.builder()
                                       .put(Environment.PATH_HOME_SETTING.getKey(), homeDir)
                                       .build();
-        final Path configDir = homeDir.resolve("config").resolve("discovery-file");
-        Files.createDirectories(configDir);
-        final Path unicastHostsPath = configDir.resolve(UNICAST_HOSTS_FILE);
+        final Path configPath;
+        if (randomBoolean()) {
```

Comment:
Can this flag be passed in, have a wrapper that continues to take a single arg for most tests, and have an explicit test to know we get the correct config dir?